Gems/LmbrCentral/Code/Source/Shape/CapsuleShape.cpp

```diff
@@ -213,31 +215,63 @@ namespace LmbrCentral
         return intersection;
     }
 
+    AZ::Vector3 CapsuleShape::GetTranslationOffset() const
+    {
+        return m_capsuleShapeConfig.m_translationOffset;
+    }
```

Comment:
The same way that `SetTranslationOffset` is protected to do nothing when shape component translation is not enabled, `GetTranslationOffset` should do the same and return offset 0 if it's disabled.

As it is now since `m_capsuleShapeConfig.m_translationOffset` is serialized then it still returns the offset value it was saved last while translation was enabled.

But I guess that's still dangerous having `m_capsuleShapeConfig.m_translationOffset` with a value and `GetTranslationOffset` returning 0. I wonder if there is a better way during reflection to ignore `m_capsuleShapeConfig.m_translationOffset` and set it to 0 if `IsShapeComponentTranslationEnabled` is false. At the moment it's only setting visibility to false, but the variable will still be there with the last saved value.

For example, `UpdateIntersectionParamsImpl` uses `m_translationOffset` directly, doesn't check if translation is enabled or not, so even if translation is disabled it can still apply an offset if it read a value from serialization.

Do we need a runtime check for `IsShapeComponentTranslationEnabled` at all? Why do we need to disable it? It's making this more complicated to manage.

Reply:
Thinking about it too it might be useful to do the check...

```
if (offsetSupported) {
    return translationOffset;
}
return zero;
```

But I'm not sure how much overhead there is to that (both from runtime and code complexity) - ultimately your call @greerdv but a good callout from @moraaar as something to consider

Reply:
>  the place where the setting is defined makes it clear that it is work in progress

That's fair enough. If that check is ultimately going to be removed once the work is completely then it's not big worry.

Reply:
Yeah, this should only be for a few weeks while this work is going on.